atlasdb-commons/src/main/java/com/palantir/common/remoting/ServiceNotAvailableException.java

```diff
@@ -29,33 +30,52 @@
     private static final long serialVersionUID = 2L;
 
     private final Optional<HostAndPort> serviceHint;
+    private final Optional<URL> serviceHintUrl;
 
     public ServiceNotAvailableException(String message, Throwable cause, HostAndPort serviceHint) {
         super(message, cause);
         this.serviceHint = Optional.of(serviceHint);
+        this.serviceHintUrl = Optional.empty();
     }
 
     public ServiceNotAvailableException(String message, HostAndPort serviceHint) {
         super(message);
         this.serviceHint = Optional.of(serviceHint);
+        this.serviceHintUrl = Optional.empty();
     }
 
     public ServiceNotAvailableException(String message, Throwable cause) {
         super(message, cause);
         this.serviceHint = Optional.empty();
+        this.serviceHintUrl = Optional.empty();
```

Comment:
don't really like this, it's somewhat awful, it's also difficult to change because `serviceHint` is used/passed in big internal product.

service hint is now bifurcated depending on which one you created the exception with, I'd prefer using `HostAndPort` over this tbh.

Reply:
Yeah, not ideal but we need a URL coming into RetryOther right? do you want to manually hardcode the https part there?

Reply:
sure, I'd rather pass the `HostAndPort` around and then establish the mapping inside the only place we want it.

Reply:
Sure, that's what it was. I didn't realize that large internal product was setting this so we can't just change the exception. In that case your plan sounds solid